google-storage/src/google-content-provider.ts

```diff
@@ -0,0 +1,216 @@
+		// Support for HTTP requests made with `Accept-Encoding: gzip`
+		gzip: true,
+		metadata: {
+			cacheControl: 'no-cache',
+		},
+	})).pipe(
+        map((result : any) => {
+            return createSuccessAjaxResponse(result[0]);
+		}),
+		catchError(error => of(createErrorAjaxResponse(404, error)))
+	);
+	response.subscribe(response => console.log(response));
+	console.log(`${filePathLocal} uploaded to ${bucketName}.`);
+	return response;
+}
+/**
+Deletes the file from the bucket
+* @param storage
+* @param bucketName
+* @param fileName
+* @returns An Observable with the response
+*/
+export function save(storage: any, bucketName: string, fileName: string): Observable < AjaxResponse > {
+	const file = storage.bucket(bucketName).file(fileName);
+	const contents = 'This is the updated contents of the file.';
+	var response= from(file.save(contents)).pipe(
+        map((result : any) => {
+            return createSuccessAjaxResponse(file);
+		}),
+		catchError(error => of(createErrorAjaxResponse(404, error)))
+	);
+	response.subscribe(response => console.log(response));
+	return response;
+}
+/**
+Deletes the file from the bucket
+* @param storage
+* @param bucketName
+* @param fileName
+* @returns An Observable with the request response
+*/
+export function remove(storage: any, bucketName: string, fileName: string): Observable < AjaxResponse > {
+	const file = storage.bucket(bucketName).file(fileName);
+	var response= from(file.delete()).pipe(
+        map((result : any) => {
+			console.log(`gs://${bucketName}/${fileName} deleted.`);
+			return createSuccessAjaxResponseForDeleteFile(result[0]);
+		}),
+		catchError(error => of(createErrorAjaxResponse(404, error)))
+	);
+	response.subscribe(response => console.log(response));
+	return response;
+}
+
+export function listCheckpoints() : Observable <AjaxResponse> {
+	throw new Error("Not implemented");
+}
+export function createCheckpoint() : Observable <AjaxResponse> {
+	throw new Error("Not implemented");
+}
+export function deleteCheckpoint() : Observable <AjaxResponse> {
+	throw new Error("Not implemented");
+}
+export function restoreFromCheckpoint() : Observable <AjaxResponse >{
+	throw new Error("Not implemented");
+}
+
+export class GoogleProvider {
+
+	// check if service account exists with correct format
+	storage : any;
+	bucketName: string;
+	fileName: string;
+	filePathLocal: string;
+
+	constructor(utility : any, bucketName: string, fileName: string, filePathLocal: string) { // Constructor
+		const serviceAccount=utility.checkServiceAccount();
+
+		var storage = new Storage.Storage({
+			projectId: serviceAccount.project_id,
+			credentials: {
+				client_email: serviceAccount.client_email,
+				private_key: serviceAccount.private_key,
+			},
+		});		
+		this.storage=storage
+		this.bucketName = bucketName;
+		this.fileName = fileName;
+		this.filePathLocal = filePathLocal;
+	}
+
+	public get(storage: any, bucketName: string, fileName: string) {
```

Comment:
Nit: Instead of invoking `get` and `update` here I would just implement the methods above inside the content provider.

There's no need to move the logic into seperate functions and it makes the code harder to read.